### PR TITLE
Update helm charts beyond `values.yaml`

### PR DESCRIPTION
**Pull request by jeffwidman**
The docker updater for helm charts was hardcoded to `"values.yaml"`.

But we should support the full range of filenames supported in the file fetcher... for example, in the wild we saw this failing to update `"arc-values.yaml"` because of this hardcoding.

### CODE REVIEW

**Comment by jeffwidman** [comment]
I broke the unit tests out separately into #6105 so that it's not blocking merging this code... I'm still learning the Rspec DSL, but since this code is behind a feature flag I'd like to get it live sooner so that if there are other bugs/issues I can see them sooner even while I work through the unit tests. Also so that I can continue to test this code w/o being blocked because other maintainers are offline for Veteran's day.

**Comment by jeffwidman** [comment]
I manually confirmed this is working against the previously failing repo:
```
[dependabot-core-dev] ~/dependabot-core $ LOCAL_GITHUB_ACCESS_TOKEN=supersecret bin/dry-run.rb --dir "/manifest_staging/charts/csi-secrets-store-provider-azure" docker jeffwidman/secrets-store-csi-driver-provider-azure --updater-options=kubernetes_updates

...

 => requirements update strategy:
 => updating oss/kubernetes-csi/secrets-store/driver from v1.2.3 to v1.2.4.1

    ± arc-values.yaml
    ~~~
    115c115
    <       tag: v1.2.3
    ---
    >       tag: v1.2.4.1
    128c128
    <         tag: v1.2.3
    ---
    >         tag: v1.2.4.1
    140c140
    <       tag: v1.2.3
    ---
    >       tag: v1.2.4.1
    ~~~

    ± values.yaml
    ~~~
    117c117
    <       tag: v1.2.3
    ---
    >       tag: v1.2.4.1
    130c130
    <         tag: v1.2.3
    ---
    >         tag: v1.2.4.1
    141c141
    <       tag: v1.2.3
    ---
    >       tag: v1.2.4.1
    ~~~
🌍 Total requests made: '0'
```

**Review by mctofu — APPROVED** [review]

**Review by honeyankit — APPROVED** [review]

**Comment by jeffwidman** [comment]
This has been deployed. I'll finish up the unit tests shortly in the other PR.

**Pull request merged**
